pkg/kubelet/network/hostport/hostport_manager_test.go

```diff
@@ -135,35 +137,37 @@ func TestHostportManager(t *testing.T) {
-		"-A KUBE-HP-4YVONL46AKYWSKS3 -m comment --comment \"pod1_ns1 hostport 8080\" -m tcp -p tcp -j DNAT --to-destination 10.1.1.2:80":  true,
-		"-A KUBE-HP-7THKRFSEH4GIIXK7 -m comment --comment \"pod1_ns1 hostport 8081\" -s 10.1.1.2/32 -j KUBE-MARK-MASQ":                    true,
-		"-A KUBE-HP-7THKRFSEH4GIIXK7 -m comment --comment \"pod1_ns1 hostport 8081\" -m udp -p udp -j DNAT --to-destination 10.1.1.2:81":  true,
-		"-A KUBE-HP-5N7UH5JAXCVP5UJR -m comment --comment \"pod3_ns1 hostport 8443\" -s 10.1.1.4/32 -j KUBE-MARK-MASQ":                    true,
-		"-A KUBE-HP-5N7UH5JAXCVP5UJR -m comment --comment \"pod3_ns1 hostport 8443\" -m tcp -p tcp -j DNAT --to-destination 10.1.1.4:443": true,
-		`COMMIT`: true,
+	linesList := strings.Split(string(raw), "\n")
+	for i, line := range linesList {
+		linesList[i] = strings.TrimSpace(line)
 	}
-	for _, line := range lines {
-		if len(strings.TrimSpace(line)) > 0 {
-			_, ok := expectedLines[strings.TrimSpace(line)]
-			assert.EqualValues(t, true, ok)
-		}
+	lines := sets.NewString(linesList...)
+
+	expectedLines := sets.NewString(
+		`*nat`,
+		`:KUBE-HOSTPORTS - [0:0]`,
+		`:KUBE-LOCALDEST - [0:0]`,
+		`:OUTPUT - [0:0]`,
+		`:PREROUTING - [0:0]`,
+		`:KUBE-HP-4YVONL46AKYWSKS3 - [0:0]`,
+		`:KUBE-HP-7THKRFSEH4GIIXK7 - [0:0]`,
+		`:KUBE-HP-5N7UH5JAXCVP5UJR - [0:0]`,
+		"-A KUBE-HOSTPORTS -m comment --comment \"pod3_ns1 hostport 8443\" -m tcp -p tcp --dport 8443 -j KUBE-HP-5N7UH5JAXCVP5UJR",
+		"-A KUBE-HOSTPORTS -m comment --comment \"pod1_ns1 hostport 8081\" -m udp -p udp --dport 8081 -j KUBE-HP-7THKRFSEH4GIIXK7",
+		"-A KUBE-HOSTPORTS -m comment --comment \"pod1_ns1 hostport 8080\" -m tcp -p tcp --dport 8080 -j KUBE-HP-4YVONL46AKYWSKS3",
+		"-A KUBE-LOCALDEST -m comment --comment \"maybe kube hostport\" -d 127.0.0.1/32 -j KUBE-HOSTPORTS",
+		"-A OUTPUT -m comment --comment \"maybe kube hostport\" -m addrtype --dst-type LOCAL -j KUBE-LOCALDEST",
+		"-A PREROUTING -m comment --comment \"maybe kube hostport\" -m addrtype --dst-type LOCAL -j KUBE-LOCALDEST",
+		"-A KUBE-HP-4YVONL46AKYWSKS3 -m comment --comment \"pod1_ns1 hostport 8080\" -s 10.1.1.2/32 -j KUBE-MARK-MASQ",
+		"-A KUBE-HP-4YVONL46AKYWSKS3 -m comment --comment \"pod1_ns1 hostport 8080\" -m tcp -p tcp -j DNAT --to-destination 10.1.1.2:80",
+		"-A KUBE-HP-7THKRFSEH4GIIXK7 -m comment --comment \"pod1_ns1 hostport 8081\" -s 10.1.1.2/32 -j KUBE-MARK-MASQ",
+		"-A KUBE-HP-7THKRFSEH4GIIXK7 -m comment --comment \"pod1_ns1 hostport 8081\" -m udp -p udp -j DNAT --to-destination 10.1.1.2:81",
+		"-A KUBE-HP-5N7UH5JAXCVP5UJR -m comment --comment \"pod3_ns1 hostport 8443\" -s 10.1.1.4/32 -j KUBE-MARK-MASQ",
+		"-A KUBE-HP-5N7UH5JAXCVP5UJR -m comment --comment \"pod3_ns1 hostport 8443\" -m tcp -p tcp -j DNAT --to-destination 10.1.1.4:443",
+		`COMMIT`)
+	for line := range expectedLines {
+		_, ok := lines[line]
+		assert.EqualValues(t, true, ok, "line %q not found in output", line)
 	}
 
```

Comment:
There are extra lines - for example, the iteration over all interfaces produces a line for whatever the local machine has.  I could go through the plumbing to inject an address-getter interface, I guess?